src/VisualStudio/Core/Def/External/UnitTesting/UnitTestingReferencesService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.VisualStudio.Language.CodeLens;
+using Microsoft.VisualStudio.Language.CodeLens.Remoting;
+using Microsoft.VisualStudio.LanguageServices.CodeLens;
+
+namespace Microsoft.CodeAnalysis.UnitTesting.ExternalAccess
+{
+    internal class UnitTestingReferencesService
+    {
+        private static readonly IEnumerable<(string MethodFullyQualifedName, string MethodFilePath, string MethodOutputFilePath)> Empty =
+            Enumerable.Empty<(string MethodFullyQualifedName, string MethodFilePath, string MethodOutputFilePath)>();
+
+        internal static async Task<IEnumerable<(string MethodFullyQualifedName, string MethodFilePath, string MethodOutputFilePath)>> GetCallerMethodsAsync(
+            IAsyncCodeLensDataPointProvider provider,
```

Comment:
@shyamnamboodiripad if we are talking through IVT, I think risk of you having some duplicated type/string might be fine? since with IVT, we can break you as much as you can break you through duplicated type and string?

I think only reason you need IVT is for "ReferenceMethodDescriptor" and "ICodeLensContext.FindReferenceMethodsAsync"

since we are talking about communication between 2 process through json, "ReferenceMethodDescriptor" doesn't need to be same type. it just need to be same shape. 

so, you could just copy and paste same type to your OOP dll and it will work fine.

another one is  nameof(ICodeLensContext.FindReferenceMethodsAsync), and you could just use "FindReferenceMethodsAsync", that that could work fine as well?

after that, you probably won't need IVT?

Reply:
Thanks for the suggestion @heejaechang. Yes although I haven't tried this what you say should be possible.

However, I think having a strongly typed helper here is better as this makes it harder for any future refactoring around "ReferenceMethodDescriptor" and "ICodeLensContext.FindReferenceMethodsAsync" to break the referencing unit testing component. For example, if someone were to add another field on ReferenceMethodDescriptor the json would change and could break consumer. Similarly with addition of arguments on or additional args on ICodeLensContext.FindReferenceMethodsAsync. With the helper, such breaks would be less likely...